packages/odf-plugin-sdk/extensions/external-storage/external-storage.ts

```diff
@@ -0,0 +1,102 @@
+import { K8sResourceKind } from '@odf/shared/types';
+import { K8sModel } from '@openshift-console/dynamic-plugin-sdk/lib/api/common-types';
+import {
+  Extension,
+  ExtensionDeclaration,
+} from '@openshift-console/dynamic-plugin-sdk/lib/types';
+import { Control } from 'react-hook-form';
+
+type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
+
+type Model = PartialBy<K8sModel, 'abbr' | 'label' | 'labelPlural'>;
+
+/**
+ *  Configures a new external storage vendor to the Storage System Creation wizard.
+ */
+export type StorageClassWizardStepProps = {
+  /** Display name of the external storage vendor. */
+  displayName: string;
+  /** The model referring the `apiGroup`,`apiVersion`, `plural` and `kind` of the external storage vendor's CRD. */
+  model: Model;
+  /** A React Functional Component to input the connection details of the external storage vendor. */
+  component: React.FunctionComponent<StorageClassComponentProps<{}>>;
+  /**  Handler function to create external storage storage vendor CR or resources. */
+  createPayload: CreatePayload<{}>;
+  /**  Handler function to validate the storage class page in order to move to the next step of wizard */
+  canGoToNextStep: CanGoToNextStep<{}>;
+  /** A function returing a promise, resolving any resources which should be created (if any) before creation of external sub-system */
+  waitToCreate?: (model: Model) => Promise<void>;
+};
+
+/** Props for `StorageClassWizardStepProps.component` to input the connection details of the external storage vendor. */
+export type StorageClassComponentProps<S extends ExternalState> = {
```

Comment:
```suggestion
export type StorageClassComponentProps<S extends ExternalState = ExternalState> = {
```
I don't see the need for generics here.